### ISSUE TITLE
[conda packaging] [windows] Adding H5_BUILT_AS_DYNAMIC_LIB to definitions

### ISSUE THREAD

**Issue by SylvainCorlay** [BlueBrain/HighFive#155]
I opened the PR to package highfive on conda-forge. This will be a dependency to xtensor-io in our next release. The PR url is: https://github.com/conda-forge/staged-recipes/pull/7195

When not disabling the build of the examples, we are seeing some errors on windows. The full logs are available at: https://ci.appveyor.com/project/conda-forge/staged-recipes/builds/20860171

To include some information about this:

```
Scanning dependencies of target read_write_single_scalar_bin
[  5%] Building CXX object src/examples/CMakeFiles/read_write_single_scalar_bin.dir/read_write_single_scalar.cpp.obj
cl : Command line warning D9002 : ignoring unknown option '-std=c++11'
read_write_single_scalar.cpp
c:\bld\highfive_1544226794732\work\include\highfive\bits/H5PropertyList_misc.hpp(82): warning C4267: 'argument': conversion from 'size_t' to 'int', possible loss of data
c:\bld\highfive_1544226794732\work\include\highfive\bits/H5DataType_misc.hpp(132): warning C4244: '=': conversion from 'double' to 'std::size_t', possible loss of data
[ 10%] Linking CXX executable read_write_single_scalar_bin.exe
LINK Pass 1: command "C:\PROGRA~2\MI0E91~1.0\VC\bin\amd64\link.exe /nologo @CMakeFiles\read_write_single_scalar_bin.dir\objects1.rsp /out:read_write_single_scalar_bin.exe /implib:read_write_single_scalar_bin.lib /pdb:%SRC_DIR%\src\examples\read_write_single_scalar_bin.pdb /version:0.0 /machine:x64 /debug /INCREMENTAL /subsystem:console %PREFIX%\Library\lib\hdf5.lib kernel32.lib user32.lib gdi32.lib winspool.lib shell32.lib ole32.lib oleaut32.lib uuid.lib comdlg32.lib advapi32.lib /MANIFEST /MANIFESTFILE:CMakeFiles\read_write_single_scalar_bin.dir/intermediate.manifest CMakeFiles\read_write_single_scalar_bin.dir/manifest.res" failed (exit code 1120) with the following output:
read_write_single_scalar.cpp.obj : MSIL .netmodule or module compiled with /GL found; restarting link with /LTCG; add /LTCG to the link command line to improve linker performance
LINK : warning LNK4075: ignoring '/INCREMENTAL' due to '/LTCG' specification
read_write_single_scalar.cpp.obj : error LNK2001: unresolved external symbol H5T_NATIVE_DOUBLE_g
read_write_single_scalar.cpp.obj : error LNK2001: unresolved external symbol H5T_NATIVE_INT_g
read_write_single_scalar_bin.exe : fatal error LNK1120: 2 unresolved externals
NMAKE : fatal error U1077: 'C:\bld\highfive_1544226794732\_build_env\Library\bin\cmake.exe' : return code '0xffffffff'
Stop.
```

By the way, if any of the highfive maintainers wants to be a maintainer of the conda recipe, let me know! I would be happy to add them in the PR.

**Comment by SylvainCorlay** [comment]
By the way, if any of the highfive maintainers wants to be a maintainer of the conda recipe, let me know! I would be happy to add them in the PR.

**Comment by SylvainCorlay** [comment]
Note: adding the definition `H5_BUILT_AS_DYNAMIC_LIB` appears to be solving that issue.

This definition should always be added to link with hdf5 with MSVC as per hdf5's documentation:

```
Users who want to build and run an application with HDF5 in Visual Studio
without using CMake should consult the USING_HDF5_VS.txt file. Building
applications with the dynamic/shared hdf5 libraries requires that the
"H5_BUILT_AS_DYNAMIC_LIB" compile definition be used.
```

**Comment by SylvainCorlay** [comment]
However, this definition appears to be exported by the HDF5 target's interface. I still need to figure out why it is not used when building highfive. Will open a clean PR when I figure this out.

**Comment by alkino** [comment]
Hi,
On the PR to make HighFive build by travis on windows I need to add this flag:
See it here: https://github.com/BlueBrain/HighFive/pull/183/files#diff-af3b638bc2a3e6c650974192a53c7291R34
So, you should be able to build highfive for conda.

**Comment by alkino** [comment]
Fix by #183
